src/common/tests/CommonUtilsUT.cpp

```diff
@@ -106,6 +106,31 @@ TEST_F(CommonUtilsTest, SavePayloadToFileInvalidArgument)
     EXPECT_FALSE(SavePayloadToFile(m_path, m_data, 0, nullptr));
 }
 
+TEST_F(CommonUtilsTest, SecureSaveToFile)
```

Comment:
If these tests are modifying the system with temp files, should they stay as part of our _unit_ tests (rather than separate functional tests)?